helper/dynamic_property_support.go

```diff
@@ -0,0 +1,85 @@
+package helper
+
+import (
+	"fmt"
+	"github.com/buildpacks/libcnb"
+	"github.com/magiconair/properties"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"os"
+	"strings"
+)
+
+const prefix = "BPL_TOMCAT_ENV_"
+
+type DynamicPropertySupport struct {
+	Bindings libcnb.Bindings
+	Logger   bard.Logger
+}
+
+func (a DynamicPropertySupport) Execute() (map[string]string, error) {
+	systemProperties := make(map[string]string)
+
+	for _, b := range a.Bindings {
```

Comment:
At present, this is looping over all bindings. We want to filter this to be a specific binding type, that way we're not looking at unrelated bindings.

See https://github.com/paketo-buildpacks/azure-application-insights/blob/main/helper/properties.go#L34

You basically do `bindings.ResolveOne` and you'll get just the binding for the specific type.

Reply:
This has been updated in the latest PR